src/test/java/net/minestom/server/entity/EntityVelocityIntegrationTest.java

```diff
@@ -102,6 +105,27 @@ public void doubleKnockback(Env env) {
         });
     }
 
+    @Test
+    public void testHasVelocity(Env env) {
+        var instance = env.createFlatInstance();
+        loadChunks(instance);
+
+        var entity = new Entity(EntityType.ZOMBIE);
+        // Should  be false because the new entity should have no velocity
+        assertFalse(entity.hasVelocity());
+
+        entity.setInstance(instance, new Pos(0, 40.1, 0)).join();
```

Comment:
Tick once after setting the instance. Entity should actually have no velocity because it just spawned and hasn't been ticked. This is returning true because of my comment above.